plugins/vpc-bridge/config/k8s_connector.go

```diff
@@ -0,0 +1,65 @@
+// Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//     http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+//go:build enablek8sconnector && windows
+
+package config
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/aws/amazon-vpc-cni-plugins/network/vpc"
+	"github.com/aws/amazon-vpc-cni-plugins/plugins/vpc-bridge/config/k8s"
+)
+
+var (
+	retrievePodConfigHandler = retrievePodConfig
+)
+
+// retrievePodConfig retrieves a pod's configuration from an external source.
+func retrievePodConfig(netConfig *NetConfig) error {
+
+	kc := netConfig.Kubernetes
+
+	// create context with timeout.
+	ctx := context.Background()
+	ctx, _ = context.WithTimeout(ctx, time.Minute)
+
```

Comment:
You should capture the cancelation and then call cancel on a defer

Reply:
done.